### PR TITLE
Edit functionality bloc

### PR DESCRIPTION
**Pull request by sharpoverride**
Convert edit widget functionality to Bloc design pattern

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:sandbox/todo/repository.dart';
```
Comment (ext-allen-ef) [inline lib/main.dart:3]: prefer relative paths > package imports

File: lib/main.dart [hunk 2]
```diff
@@ -51,116 +51,82 @@ class Home extends StatelessWidget {
...
+                });
+          }
+          if (state is AddedTodoFormState) {
+            context.read<TodoBloc>().add(AddTodoEvent(todo: state.todo));
+          }
+
+          if(state is SavedTodoFormState) {
+            context.read<TodoBloc>().add(UpdateTodoEvent(newTodo: state.todo, toBeUpdated: state.toBeUpdated));
+          }
+          if (state is AddedTodoFormState || state is SavedTodoFormState) {
```
Comment (ext-allen-ef) [inline lib/main.dart:76]: Popping navigator from `main.dart` seems a bit confusing to me, maybe better to move this to the TodoFormWidget

File: lib/main.dart [hunk 3]
```diff
@@ -51,116 +51,82 @@ class Home extends StatelessWidget {
...
+              todos: state.all,
+            );
           }
           return Text('Unhandled state');
         }),
       ),
       floatingActionButton: FloatingActionButton(
-        onPressed: () =>
-            context.read<TodoFormBloc>().add(ShowAddTodoFormEvent()),
+        onPressed: () async {
```
Comment (ext-allen-ef) [inline lib/main.dart:110]: why async?

File: lib/navigation/event.dart [hunk 4]
```diff
@@ -1,9 +1,11 @@
-import 'model.dart';
+import 'navigation.dart';
 
-abstract class NavigationEvent {}
+class NavigationEvent {
+  Navigation navigation = Navigation.all;
+}
 
 class UpdateNavigationEvent extends NavigationEvent {
-  final Navigation navigation;
```
Comment (ext-allen-ef) [inline lib/navigation/event.dart:6]: why remove final?

File: lib/navigation/state.dart [hunk 5]
```diff
@@ -1,9 +1,18 @@
-import 'model.dart';
+import 'navigation.dart';
 
-abstract class NavigationState {}
+class NavigationState {
+  final Navigation? navigation;
```
Comment (ext-allen-ef) [inline lib/navigation/state.dart:4]: why are these optional?

File: lib/todo/bloc.dart [hunk 6]
```diff
@@ -1,24 +1,41 @@
 import 'package:bloc/bloc.dart';
+import 'package:sandbox/todo/model.dart';
```
Comment (ext-allen-ef) [inline lib/todo/bloc.dart:2]: prefer relative imports: https://dart.dev/guides/language/effective-dart/usage#prefer-relative-import-paths

File: lib/todo/repository.dart [hunk 7]
```diff
@@ -1,18 +1,26 @@
...
+    return todos.where((x) => x.completed).toList();
+  }
+
+  stillTodos() {
+    return todos.where((x) => !x.completed).toList();
+  }
+
   void addTodo(Todo todo) {
-    todos.add(todo);
+    todos.add(Todo(name: todo.name, completed: todo.completed));
```
Comment (ext-allen-ef) [inline lib/todo/repository.dart:19]: prefer to use copy with, or i  think you can just assign directly, shouldn't be an issue.

File: lib/todo/state.dart [hunk 8]
```diff
@@ -1,10 +1,12 @@
-import 'model.dart';
+import 'package:sandbox/todo/model.dart';
```
Comment (ext-allen-ef) [inline lib/todo/state.dart:1]: should be relative

File: lib/main.dart [hunk 9]
```diff
@@ -51,116 +51,82 @@ class Home extends StatelessWidget {
...
+          }
+          if(state is ShowEditTodoFormState) {
+            final todo = state.todo;
+            final bloc = context.read<TodoFormBloc>();
+            showDialog(
+                context: context,
+                builder: (BuildContext context) {
+                  return BlocProvider.value(
+                      value: bloc,
+                      child: AlertDialog(content: TodoFormWidget(todo: todo)));
```
Comment (KRietz) [inline lib/main.dart:66]: Add a listener here

**Pull request closed without merging**